test/es-module/test-esm-loader-spawn-promisified.mjs

```diff
@@ -0,0 +1,272 @@
+      '--no-warnings',
+      '--experimental-loader',
+      fixtures.fileURL('/es-module-loaders/hooks-custom.mjs'),
+      '--input-type=module',
+      '--eval',
+      'import \'esmHook/format.false\'',
+    ]);
+
+    assert.strictEqual(code, 1);
+    assert.strictEqual(signal, null);
+    assert.strictEqual(stdout, '');
+    assert.match(stderr, /ERR_INVALID_RETURN_PROPERTY_VALUE/);
+  });
+  it('throws on boolean true', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--experimental-loader',
+      fixtures.fileURL('/es-module-loaders/hooks-custom.mjs'),
+      '--input-type=module',
+      '--eval',
+      'import \'esmHook/format.true\'',
+    ]);
+
+    assert.strictEqual(code, 1);
+    assert.strictEqual(signal, null);
+    assert.strictEqual(stdout, '');
+    assert.match(stderr, /ERR_INVALID_RETURN_PROPERTY_VALUE/);
+  });
+
+  it('throws on invalid returned object', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--experimental-loader',
+      fixtures.fileURL('/es-module-loaders/hooks-custom.mjs'),
+      '--input-type=module',
+      '--eval',
+      'import \'esmHook/badReturnFormatVal.mjs\'',
+    ]);
+
+    assert.strictEqual(code, 1);
+    assert.strictEqual(signal, null);
+    assert.strictEqual(stdout, '');
+    assert.match(stderr, /ERR_INVALID_RETURN_PROPERTY_VALUE/);
+  });
+
+  it('throws on unsupported format', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--experimental-loader',
+      fixtures.fileURL('/es-module-loaders/hooks-custom.mjs'),
+      '--input-type=module',
+      '--eval',
+      'import \'esmHook/unsupportedReturnFormatVal.mjs\'',
+    ]);
+
+    assert.strictEqual(code, 1);
+    assert.strictEqual(signal, null);
+    assert.strictEqual(stdout, '');
+    assert.match(stderr, /ERR_UNKNOWN_MODULE_FORMAT/);
+  });
+
+  it('throws on invalid format property type', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--experimental-loader',
+      fixtures.fileURL('/es-module-loaders/hooks-custom.mjs'),
+      '--input-type=module',
+      '--eval',
+      'import \'esmHook/badReturnSourceVal.mjs\'',
+    ]);
+
+    assert.strictEqual(code, 1);
+    assert.strictEqual(signal, null);
+    assert.strictEqual(stdout, '');
+    assert.match(stderr, /ERR_INVALID_RETURN_PROPERTY_VALUE/);
+  });
+
+  it('rejects dynamic imports for all of the error cases checked above', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--experimental-loader',
+      fixtures.fileURL('/es-module-loaders/hooks-custom.mjs'),
+      '--input-type=module',
+      '--eval',
+      `import assert from 'node:assert';
+      Promise.allSettled([
```

Comment:
We should either use TLA, or wrap the then handler in `mustCall()` (otherwise we can't detect never-settling promises)
```suggestion
      await Promise.allSettled([
```